lgc/builder/YCbCrConverter.cpp

```diff
@@ -326,64 +328,78 @@ void YCbCrConverter::genSamplerDescChroma() {
 // =====================================================================================================================
 // Generate image descriptor for chroma channel
 void YCbCrConverter::genImgDescChroma() {
-  Value *pInt32One = ConstantInt::get(m_builder->getInt32Ty(), 1);
   SqImgRsrcRegHandler proxySqRsrcRegHelper(m_builder, m_imgDescLuma, m_gfxIp);
-  YCbCrAddressHandler addrHelper(m_builder, &proxySqRsrcRegHelper, m_gfxIp);
-  unsigned bpp = (m_metaData.word1.planes == 1)
-                     ? (m_metaData.word2.bitCounts.xBitCount + m_metaData.word2.bitCounts.yBitCount +
-                        m_metaData.word2.bitCounts.zBitCount + m_metaData.word2.bitCounts.wBitCount)
-                     : m_metaData.word2.bitCounts.xBitCount;
-  addrHelper.genHeightAndPitch(m_metaData.word0.bitDepth.channelBitsR, bpp, m_metaData.word2.bitCounts.xBitCount,
-                               m_metaData.word1.tileOptimal, m_metaData.word1.planes);
-  addrHelper.genBaseAddress(m_metaData.word1.planes);
-
   Value *width = proxySqRsrcRegHelper.getReg(SqRsrcRegs::Width);
   Value *height = proxySqRsrcRegHelper.getReg(SqRsrcRegs::Height);
-  Value *widthHalf = m_builder->CreateLShr(width, pInt32One);
-  Value *heightHalf = m_builder->CreateLShr(height, pInt32One);
   m_width = m_builder->CreateUIToFP(width, m_builder->getFloatTy());
   m_height = m_builder->CreateUIToFP(height, m_builder->getFloatTy());
 
-  Value *pPitch = proxySqRsrcRegHelper.getReg(SqRsrcRegs::Pitch);
-  Value *pitchHalf = m_builder->CreateLShr(pPitch, pInt32One);
+  if (m_metaData.word1.planes == 1) {
+    Value *imgDataFmt = proxySqRsrcRegHelper.getReg(SqRsrcRegs::Format);
+    Value *dstSelXYZW = proxySqRsrcRegHelper.getReg(SqRsrcRegs::DstSelXYZW);
+    Value *isGBGRFmt = nullptr;
+    Value *isBGRGFmt = nullptr;
+
+    switch (m_gfxIp->major) {
+    case 9: {
+      isGBGRFmt = m_builder->CreateICmpEQ(
+          imgDataFmt, m_builder->getInt32(Pal::Gfx9::Chip::IMG_DATA_FORMAT::IMG_DATA_FORMAT_BG_RG__CORE));
```

Comment:
We shouldn't use any PAL definitions in LLPC. Please add what you want to this (BuilderImpl.h):
```
  enum ImgDataFormat {
    IMG_DATA_FORMAT_32 = 4,
    IMG_DATA_FORMAT_32_32 = 11,
    IMG_DATA_FORMAT_32_32_32_32 = 14,
  };
```